internal/dag/cache.go

```diff
@@ -528,7 +518,7 @@ func isRefToSecret(ref gatewayapi_v1beta1.SecretObjectReference, secret *v1.Secr
 
 // LookupSecret returns a Secret if present or nil if the underlying kubernetes
 // secret fails validation or is missing.
-func (kc *KubernetesCache) LookupSecret(name types.NamespacedName, validate func(*v1.Secret) error) (*Secret, error) {
+func (kc *KubernetesCache) LookupSecret(name types.NamespacedName, validate func(*Secret) error) (*Secret, error) {
```

Comment:
I'm considering splitting this method out into `LookupTLSSecret`, `LookupCASecret`, and `LookupCRLSecret` and getting rid of the indirection of the passed-in validation function, thoughts? IMO it's a little simpler to navigate that way.

Reply:
Went ahead with this in https://github.com/projectcontour/contour/pull/4788/commits/849d3f6aa8c8bc3000c4e2b3de50018952db53e4

Reply:
yeah, i like the option you chose, also makes it a little clearer in reading calling code what exactly is happening etc.